ddtrace/internal/utils/cache.py

```diff
@@ -38,23 +39,27 @@ def get(self, key, f):  # type: ignore[override]
         function ``f`` is called on the key to generate it. The return value is
         then stored in the cache and returned to the caller.
         """
-        if len(self) >= self.maxsize:
-            for _, h in zip(range(self.maxsize >> 1), sorted(self, key=lambda h: self[h][1])):
-                del self[h]
 
         _ = super(LFUCache, self).get(key, miss)
         if _ is not miss:
-            value, count = _
-            self[key] = (value, count + 1)
+            with self.count_lock:
+                value, count = _
+                self[key] = (value, count + 1)
             return value
 
         with self.lock:
             _ = super(LFUCache, self).get(key, miss)
```

Comment:
Just curious why do we need to get the cached value again. Can we just use the result from line 43?

Reply:
If the function is slow and we have a concurrent access to the same key, the first call to acquire the lock computes the value, so the second can check again when it gets the lock, instead of recomputing the value itself.